### PR TITLE
#631@patch: Handle the click on HTMLInputElement checkbox, radio and submit.

### LINKED ISSUE #631: HD doesn't recognize the checked attr in the input tag
**Describe the bug**
Not working with input checked={isChecked} prop in a react proj.

**To Reproduce**
```
const Toggle = ({
  value,
  checked = false,
  onChange,
  text
}: Props) => {
  const [checked, setChecked] = useState(false);

  const onToggle = () => {
    if (!disabled) {
      if (onChange) onChange();
      setChecked(prev => !prev);
    }
  };

  return (
    <div className="toggle" data-testid="toggle">
      <label data-testid="toggle-label" className="toggle-switch">
        <input
          data-testid="toggle-input"
          value={value}
          type="checkbox"
          checked={checked}
          onChange={onToggle}
        />
      </label>
    </div>
  );
};
```

**Expected behavior**
**PASS THIS TEST:** 
```
render(<Toggle onChange={handleChange} text={someText} value={someVal} />);

    expect(screen.getByTestId("toggle")).toBeInTheDocument();
    expect(screen.getByTestId("toggle-input")).not.toBeChecked();

    await userEvent.click(screen.getByTestId("toggle-label"));

    expect(handleChange).toBeCalledTimes(1); //FAILS
    expect(screen.getByTestId("toggle-input")).toBeChecked();  //FAILS
```

**Screenshots**

**Device:**
 - OS: macOs 12.6
 - Browser [chrome]
 - Version [106]

**Additional context**
I tried to use vitest with HD, but it didn't work until I switched to jsdom

### PR DESCRIPTION
**Pull request by mehdi-valette**
This patch includes the implementation of onclick for HTMLInputElement and the unit tests that go with it.

More precisely :

For checkboxes: toggle the attribute "checked" and trigger onchange.

For radio: if the radio was select -> nothing. Otherwise, selected the clicked radio, deselect other radios in the same form with the same name, and trigger onchange.

For submit: trigger the onsubmit of the parent form.

I discovered happy-dom recently, I hope the code syntax is appropriate.

### CODE REVIEW

File: packages/happy-dom/test/nodes/html-input-element/HTMLInputElement.test.ts [hunk 1]
```diff
@@ -584,4 +584,103 @@ describe('HTMLInputElement', () => {
...
+				cancelable: true
+			});
+			expectedEvent._target = form;
+			expectedEvent._currentTarget = form;
+
+			const submitted = jest.fn();
+			form.addEventListener('submit', submitted);
+
+			submit.click();
+			expect(submitted).toHaveBeenCalledTimes(1);
```
Comment (capricorn86) [inline packages/happy-dom/test/nodes/html-input-element/HTMLInputElement.test.ts:683]:
```suggestion
			
			let triggeredEvent: Event = null;
			form.addEventListener('submit', (event) => triggeredEvent = event);

			submit.click();
			
			expect(triggeredEvent.target === form).toBe(true);
			expect(triggeredEvent.currentTarget === form).toBe(true);

```

**Review by capricorn86 — CHANGES_REQUESTED** [review]
Thank you for your contribution! :star2: 

I have found some minor things that we should fix. Otherwise it looks good :+1: 

Please also note that there is an ongoing PR that will improve the "checked" property behaviour:
https://github.com/capricorn86/happy-dom/pull/677/files#diff-bb649479cc3389b9fff9883a1cc18a2b9edc08aac12cb0689ecabf13f21be16bR568

File: packages/happy-dom/src/nodes/html-input-element/HTMLInputElement.ts [hunk 2]
```diff
@@ -52,6 +52,48 @@ export default class HTMLInputElement extends HTMLElement implements IHTMLInputE
...
+				this.setAttributeNS(null, 'checked', '');
+
+				dispatchEvent();
+
+				break;
+
+			case 'submit':
+				dispatchEvent('submit', this.form);
+		}
+	};
```
[the remark this answers is not in the document]
Comment (mehdi-valette) [reply packages/happy-dom/src/nodes/html-input-element/HTMLInputElement.ts:96]: That's much simpler! If I knew that 😅

Just a thought for the line 59  `this.checked = !this.checked`. Do we want it for the radio? If I click on a checked radio it shouldn't uncheck. (Unless it is already handled at another place in the code).

[the remark this answers is not in the document]
Comment (capricorn86) [reply packages/happy-dom/src/nodes/html-input-element/HTMLInputElement.ts:96]: I have corrected the code :slightly_smiling_face:

**Comment by capricorn86** [comment]
@mehdi-valette I believe that support for this has been added in this PR:
https://github.com/capricorn86/happy-dom/pull/677/files#diff-bb649479cc3389b9fff9883a1cc18a2b9edc08aac12cb0689ecabf13f21be16bR1042

I will close this PR.

**Pull request closed without merging**